src/Stratis.Bitcoin/Connection/ConnectionManager.cs

```diff
@@ -134,6 +134,9 @@ public IReadOnlyNetworkPeerCollection ConnectedPeers
 
         public List<NetworkPeerServer> Servers { get; }
 
+        /// <summary>Maintains a list of connected peers and ensures their proper disposal.</summary>
+        private readonly ConnectedPeersHelper connectedPeersHelper;
```

Comment:
I think that would be a bit misleading since we already have a `NetworkPeerCollection`. But maybe functionality of the helper can be moved to a `NetworkPeerCollection` class.

Also I don't think that `Collection` will be an appropriate name for the functionality presented in the helper since it doesn't implement `IEnumerable`

Reply:
At 1st we called that network peer disposed helper, because its primary objective was to take care of network peer disposals. However it also maintains a list of all connected peers, at which point the 1st name we selected seemed an appropriate.

Then as no escape says, there is network peer collection already which is used by connection manager, which purpose is to provide a collection of connected peers all other components. For example if another component was to set a message to all connected peers it should use connection managers peer collection.

This is why we did not want to name it is a collection because then we would have 2 collections with completely different purpose and it could be confusing which is which. So this helper class can  be used by any component that actually creates new peers. If a company does that did no longer it needs to track those newly created peers itself and care about their disposal.

So this is the purpose of this class, feel free to come up with better name, we struggled with that ourselves. I agree that since it doesn't really implement enumeration, it is probably not a collection anyway.

Reply:
I don't think the functionality of this class should be part of the collection, especially when considering that the collection for connection manager must not dispose the peers.

Reply:
How about something like PeerLifetime or PeerDisposeMonitor etc...
Also instead of using a CD you could just inject the exiting `NetworkPeerCollection` and not need to mange in internal list.

Reply:
Why would we use a heavier collection for something we only need dictionary for? That doesn't make sense.

As explained above no longer this component is only about disposing, it also tracks all connected peers of the owning component. So peer dispose monitor is not very good I think.

Peer lifetime also doesn't resonate with me. But in general I agree that we did struggle with naming this class. But so far no better ideas in my head.

Any ideas @noescape00?

Reply:
`PeerDisposalHelper` as it was before, maybe?